python/paddle/tensor/math.py

```diff
@@ -7728,3 +7728,70 @@ def signbit(x, name=None):
     x = paddle.sign(neg_zero_x)
     out = paddle.cast(x < 0, dtype='bool')
     return out
+
+
+def sinc(x, name=None):
+    r"""
+    Calculate the normalized sinc of ``x`` elementwise.
+
+    .. math::
+
+        out_i =
+        \left\{
+        \begin{aligned}
+        &1 & \text{ if $x_i = 0$} \\
+        &\frac{\sin(\pi x_i)}{\pi x_i} & \text{ otherwise}
+        \end{aligned}
+        \right.
+
+    Args:
+        x (Tensor): The input Tensor. Must be one of the following types: float16, float32, float64.
+        name (str, optional): Name for the operation (optional, default is None). For more information, please refer to :ref:`api_guide_Name`.
+
+    Returns:
+        out (Tensor): The Tensor of elementwise computed normalized sinc result.
+    """
+    if not isinstance(x, (paddle.Tensor, Variable, paddle.pir.Value)):
+        raise TypeError(f"x must be tensor type, but got {type(x)}")
+
+    check_variable_and_dtype(
+        x,
+        "x",
+        [
+            'float16',
+            'float32',
+            'float64',
+        ],
+        "sinc",
+    )
+
+    tmp = paddle.where(x != 0, x, paddle.full_like(x, 1.0e-20))
```

Comment:
why use `1.0e-20`, if data type of `x` is float16, will there be underflow?

Reply:
Can we support backwarc gradient calculation, especially when x=0?

Reply:
shall we add test case for calculating gradient

Reply:
> Can we support backward gradient calculation, especially when x=0?

when `x=0`，the gradient should be 0:  

$$( \frac{\pi x}{\sin(\pi x)} )^{\prime} = \pi \frac{\sin(\pi x) - \pi x \cos(\pi x)}{\sin^2(\pi x)}$$

by LHopital's rule,

$$ \lim_{x \rightarrow 0} \frac{\sin(\pi x) - \pi x \cos(\pi x)}{\sin^2(\pi x)} = \lim_{x \rightarrow 0} \frac{\pi \cos(\pi x) - [\pi \cos(\pi x) - \pi^2 x \sin(\pi x) ]}{2 \pi \sin(\pi x) \cos(\pi x)} =  \lim_{x \rightarrow 0} \frac{\pi x}{2 \pi cos(\pi x)} = 0$$

The torch example:
```python
x = torch.rand(2,3)
x[0,0] = 0.0
x.requires_grad = True
s = torch.sinc(x).sum()
s.backward()
x.grad
# tensor([[ 0.0000, -1.2824, -1.0302],
#        [-1.3687, -1.0435, -1.1842]])
```

Reply:
> shall we add test case for calculating gradient

I will add it

Reply:
> Can we support backward gradient calculation, especially when x=0?

> Can we support backward gradient calculation, especially when x=0?

when `x=0`，the gradient should be 0:  

$$( \frac{\sin(\pi x)}{\pi x} )^{\prime} = \pi \frac{\pi x \cos(\pi x) - \sin(\pi x)}{\pi^2 x^2}$$

by LHopital's rule,

$$ \lim_{x \rightarrow 0} \frac{\pi x \cos(\pi x) - \sin(\pi x)}{\pi^2 x^2} = \lim_{x \rightarrow 0} \frac{ [\pi \cos(\pi x) - \pi^2 x \sin(\pi x) ] - \pi \cos(\pi x)}{2 \pi^2 x} =  \lim_{x \rightarrow 0} -\frac{\sin(\pi x)}{2} = 0$$

The torch example:
```python
x = torch.rand(2,3)
x[0,0] = 0.0
x.requires_grad = True
s = torch.sinc(x).sum()
s.backward()
x.grad
# tensor([[ 0.0000, -1.2824, -1.0302],
#        [-1.3687, -1.0435, -1.1842]])
```
The current implementation using `paddle.where` can return gradient value 0 if x = 0

Reply:
> Can we support backward gradient calculation, especially when x=0?

Should inplace operation `sinc_` also support gradient calculation?